pygdf/series.py

```diff
@@ -409,6 +412,40 @@ def dtype(self):
         """dtype of the Series"""
         return self._impl.dtype
 
+    @classmethod
+    def _concat(cls, objs, index=True):
+        head = objs[0]
+        for o in objs:
+            if o._impl != head._impl:
+                raise ValueError("All series must be of same type")
+
+        data = head._impl.concat(objs)
```

Comment:
The `self._data.dtype` (`Buffer.dtype`) will always have the physical dtype.  The `SeriesImpl` has a logical meaning of things.  Here, we will only need physical info.